### PR TITLE
Profile Page Backend

### LINKED ISSUE #31: Profile page
Contributors should be able to view their profile page and edit the information on their profile page.

Frontend:
- [ ] Editable: name, preferred name, gender, pronouns, date joined, phone number, LinkedIn link, website link, Twitter username, my topics
- [ ] Not editable: position, email, my roles
- [ ] A contributor should be able to edit and save changes to their own profile
- [ ] A badge should be displayed if contributor is on the masthead

Backend:
- [ ] An endpoint that saves the user's name, preferred name, gender, pronouns, date joined, phone number, LinkedIn link, website link, Twitter username, and my topics

Refs #7 #17 
[Figma mockup](https://www.figma.com/file/bZJickwpVfHvxeqxE6hRU9/SSW-P0?node-id=132%3A127)

### LINKED ISSUE #49: Integrate frontend/backend for profile page
Integrate the frontend and backend for the profile page. Use the PUT user endpoint that has been created.

Refs #7 #17

### PR DESCRIPTION
**Pull request by ajain1921**
## Summary

Profile Page Backend
Closes #49 , Closes #31 

## Changes

- Add API Calls to get and update a user

## Requests / Responses

**Request**
(Only for backend)
GET `/api/users/:userId` Returns a specific user

**Response**
(Only for backend)
HTTP/1.1 200 OK

```
{
    "success": true,
    "result": {
        "firstName": "Alice",
        "lastName": "Fang",
        "preferredName": "Alice",
        "email": "af@gmail.com",
        "phone": "732-732-7327",
        "oauthID": "2",
        "genders": [
            "Woman"
        ],
        "pronouns": [
            "She/her"
        ],
        "masthead": true,
        "onboardingStatus": "ONBOARDING_SCHEDULED",
        "portfolio": "alice.com",
        "linkedIn": "www.linkedin.com/alice",
        "twitter": "@alice",
        "claimedPitches": [
            "603ad2ab2b8abc0b23177701",
            "603adf6c838631111ec35c2e",
            "603adf9b838631111ec35c2f"
        ],
        "submittedPitches": [],
        "currentTeams": [
            "EDITING"
        ],
        "role": "CONTRIBUTOR",
        "races": [],
        "interests": [
            "CANNABIS",
            "HOUSING",
            "STAGE AND SCREEN",
            "POLITICS"
        ],
        "_id": "6031a866c70ec705736a79e5",
        "gender": null,
        "dateJoined": "2000-11-09T00:00:00.000Z",
        "__v": 3
    },
    "message": "Successfully retrieved user"
}
```
**Request**
(Only for backend)
POST `/api/users/:userId` Updates a specific user

**Response**
(Only for backend)
HTTP/1.1 200 OK
```
{
  "success": true,
  "message": "Successfully updated user",
  "result": {
    "firstName": "Alice",
    "lastName": "Fang",
    "preferredName": "Alice",
    "email": "af@gmail.com",
    "phone": "732-732-7327",
    "oauthID": "2",
    "genders": [
      "Woman"
    ],
    "pronouns": [
      "She/her"
    ],
    "masthead": true,
    "onboardingStatus": "ONBOARDING_SCHEDULED",
    "portfolio": "www.alice.com",
    "linkedIn": "www.linkedin.com/alice",
    "twitter": "www.twitter.com/alice",
    "claimedPitches": [
      "603ad2ab2b8abc0b23177701",
      "603adf6c838631111ec35c2e",
      "603adf9b838631111ec35c2f"
    ],
    "submittedPitches": [],
    "currentTeams": [
      "EDITING"
    ],
    "role": "CONTRIBUTOR",
    "races": [],
    "interests": [
      "CANNABIS",
      "HOUSING",
      "STAGE AND SCREEN",
      "POLITICS"
    ],
    "_id": "6031a866c70ec705736a79e5",
    "gender": null,
    "dateJoined": "2000-11-09T00:00:00.000Z",
    "__v": 3
  }
}
```

### CODE REVIEW

**Review by sahithimuthyala — DISMISSED** [review]
Great work! Should be good to merge once you remove comments and Alice approves too 🌟

**Review by alicesf2 — APPROVED** [review]
Wow, amazing work Aditya 🎉 I'm approving because the comments I left were about pretty small, non-blocking things but make sure you address them before you merge. A few things to note (mostly for myself, but good for you to know as well):

- [ ] I'm going to extract the hard-coded button colors to the `enums.ts` file
- [ ] I'm going to remove the null checks once I change the default schema values for `user.ts`

File: client/src/pages/profile/Profile.tsx
Comment (sahithimuthyala) [reply client/src/pages/profile/Profile.tsx:20]: Nope, Aditya mentioned that he forgot to remove them 😅

**Comment by alicesf2** [comment]
The error message should display a more user-friendly message, smth like "You must fill out all required fields" instead of "POST_PROFILE_FAIL" - this is more like an error message for the backend. But we'll add in form validation later in a separate PR anyways, approving for now so we can get more functionality in our main repo first 👍

**Pull request merged**
